cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityTopicOperator.java

```diff
@@ -246,6 +248,7 @@ public static EntityTopicOperator fromCrd(Reconciliation reconciliation, Kafka k
                 if (topicOperatorSpec.getLivenessProbe() != null) {
                     result.setLivenessProbe(topicOperatorSpec.getLivenessProbe());
                 }
+                result.setKafkaVersion(kafkaAssembly.getSpec().getKafka().getVersion());
```

Comment:
Does this in any way depend also on the `inter.broker.protocol.version`? Also, how but risk is the asynchronicity of the upgrade (Kafka is rolled first).

Reply:
I do not think it is dependent. TO uses Kafka version just to load the related config model for topics.